tests/unit/test_agent_loop_tool_calls.py

```
@@ -0,0 +1,117 @@
+
+
+class _StubLLM:
+    """Replays a fixed sequence of completions so the agent loop is deterministic."""
+
+    SUPPORTS_COMPLETION = True
+
+    def __init__(self, responses: list[ChatCompletion]) -> None:
+        self.responses = responses
+        self.calls = 0
+
+    async def acompletion(self, **kwargs: Any) -> ChatCompletion:
```

Ruff (0.16.1)

[warning] Unused method argument: `kwargs`

(ARG002)